### PR TITLE
Use int64 offset types for accessing code-points in nvtext::normalize

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Changes some internal offset arrays used for managing temporary unicode code-points to int64 type.
This effects the nvtext normalize and subword-tokenizer functions.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/text/subword/data_normalizer.cu [hunk 1]
```diff
@@ -296,20 +297,22 @@ uvector_pair data_normalizer::normalize(char const* d_strings,
...
+  auto const bytes_count = d_strings_offsets->element(input.size(), stream);
   if (bytes_count == 0) {  // if no bytes, nothing to do
     return uvector_pair{std::make_unique<rmm::device_uvector<uint32_t>>(0, stream),
-                        std::make_unique<rmm::device_uvector<cudf::size_type>>(0, stream)};
+                        std::make_unique<rmm::device_uvector<int64_t>>(0, stream)};
   }
 
-  cudf::detail::grid_1d const grid{bytes_count, THREADS_PER_BLOCK, 1};
-  size_t const threads_on_device  = grid.num_threads_per_block * grid.num_blocks;
+  int64_t const threads_per_block = THREADS_PER_BLOCK;
```
Comment (bdice) [inline cpp/src/text/subword/data_normalizer.cu:306]: Why is this one `int64_t` and not `size_t`?

**Review by bdice — APPROVED** [review]

File: cpp/src/text/subword/data_normalizer.cu (hunk 1 above)
Comment (davidwendt) [reply cpp/src/text/subword/data_normalizer.cu:306]: Because the `cudf::util::div_rounding_up_safe` requires both of its parameters to be the same type.

**Comment by davidwendt** [comment]
Benchmarks showed no significant changes:
```
# normalize

## [0] Quadro GV100

|  row_width  |  num_rows  |    type    |   Ref Time |   Cmp Time |         Diff |   %Diff |
|-------------|------------|------------|------------|------------|--------------|---------|
|     32      |    4096    |   spaces   |  81.431 us |  85.301 us |     3.870 us |   4.75% |
|     64      |    4096    |   spaces   | 115.967 us | 115.662 us |    -0.306 us |  -0.26% |
|     128     |    4096    |   spaces   | 203.274 us | 203.615 us |     0.341 us |   0.17% |
|     256     |    4096    |   spaces   | 452.965 us | 452.399 us |    -0.566 us |  -0.12% |
|     512     |    4096    |   spaces   | 870.143 us | 870.611 us |     0.468 us |   0.05% |
|    1024     |    4096    |   spaces   |   1.797 ms |   1.798 ms |     1.230 us |   0.07% |
|     32      |   32768    |   spaces   |  82.834 us |  83.505 us |     0.672 us |   0.81% |
|     64      |   32768    |   spaces   | 131.560 us | 132.233 us |     0.672 us |   0.51% |
|     128     |   32768    |   spaces   | 243.079 us | 243.760 us |     0.680 us |   0.28% |
|     256     |   32768    |   spaces   | 517.193 us | 517.805 us |     0.612 us |   0.12% |
|     512     |   32768    |   spaces   |   1.074 ms |   1.073 ms |    -1.217 us |  -0.11% |
|    1024     |   32768    |   spaces   |   2.198 ms |   2.197 ms |    -1.411 us |  -0.06% |
|     32      |   262144   |   spaces   | 176.773 us | 177.279 us |     0.507 us |   0.29% |
|     64      |   262144   |   spaces   | 390.972 us | 391.422 us |     0.450 us |   0.12% |
|     128     |   262144   |   spaces   | 987.215 us | 998.977 us |    11.762 us |   1.19% |
|     256     |   262144   |   spaces   |   4.645 ms |   4.622 ms |   -22.886 us |  -0.49% |
|     512     |   262144   |   spaces   |  11.163 ms |  11.167 ms |     4.310 us |   0.04% |
|    1024     |   262144   |   spaces   |  20.536 ms |  20.674 ms |   138.405 us |   0.67% |
|     32      |  2097152   |   spaces   | 854.996 us | 856.010 us |     1.015 us |   0.12% |
|     64      |  2097152   |   spaces   |   2.267 ms |   2.267 ms |    -0.184 us |  -0.01% |
|     128     |  2097152   |   spaces   |   5.539 ms |   5.540 ms |     1.724 us |   0.03% |
|     256     |  2097152   |   spaces   |  18.357 ms |  18.341 ms |   -15.741 us |  -0.09% |
|     512     |  2097152   |   spaces   |  49.057 ms |  49.151 ms |    93.794 us |   0.19% |
|     32      |  16777216  |   spaces   |   6.287 ms |   6.285 ms |    -1.718 us |  -0.03% |
|     64      |  16777216  |   spaces   |  17.231 ms |  17.230 ms |    -0.853 us |  -0.00% |
|     32      |    4096    | characters | 393.456 us | 390.471 us |    -2.985 us |  -0.76% |
|     64      |    4096    | characters | 420.148 us | 419.477 us |    -0.671 us |  -0.16% |
|     128     |    4096    | characters | 488.615 us | 487.908 us |    -0.707 us |  -0.14% |
|     256     |    4096    | characters | 643.055 us | 645.345 us |     2.290 us |   0.36% |
|     512     |    4096    | characters |   1.074 ms |   1.103 ms |    28.766 us |   2.68% |
|    1024     |    4096    | characters |   1.779 ms |   1.815 ms |    35.698 us |   2.01% |
|     32      |   32768    | characters | 440.692 us | 436.934 us |    -3.758 us |  -0.85% |
|     64      |   32768    | characters | 543.702 us | 541.908 us |    -1.794 us |  -0.33% |
|     128     |   32768    | characters | 748.311 us | 749.589 us |     1.278 us |   0.17% |
|     256     |   32768    | characters |   1.147 ms |   1.150 ms |     3.769 us |   0.33% |
|     512     |   32768    | characters |   1.917 ms |   1.931 ms |    14.646 us |   0.76% |
|    1024     |   32768    | characters |   3.528 ms |   3.543 ms |    14.990 us |   0.42% |
|     32      |   262144   | characters |   1.270 ms |   1.297 ms |    27.198 us |   2.14% |
|     64      |   262144   | characters |   3.023 ms |   3.002 ms |   -21.368 us |  -0.71% |
|     128     |   262144   | characters |   8.972 ms |   8.936 ms |   -36.779 us |  -0.41% |
|     256     |   262144   | characters |  24.772 ms |  24.651 ms |  -120.789 us |  -0.49% |
|     512     |   262144   | characters |  55.773 ms |  56.005 ms |   232.549 us |   0.42% |
|    1024     |   262144   | characters | 118.395 ms | 118.498 ms |   103.045 us |   0.09% |
|     32      |  2097152   | characters |   8.077 ms |   8.205 ms |   127.791 us |   1.58% |
|     64      |  2097152   | characters |  23.360 ms |  23.343 ms |   -17.133 us |  -0.07% |
|     128     |  2097152   | characters |  76.116 ms |  75.252 ms |  -863.717 us |  -1.13% |
|     256     |  2097152   | characters | 217.698 ms | 216.167 ms | -1530.771 us |  -0.70% |
|     512     |  2097152   | characters | 493.366 ms | 490.911 ms | -2455.858 us |  -0.50% |
|     32      |  16777216  | characters |  61.473 ms |  62.364 ms |   890.987 us |   1.45% |
|     64      |  16777216  | characters | 178.440 ms | 179.409 ms |   969.043 us |   0.54% |
|     32      |    4096    |  to_lower  | 390.470 us | 387.860 us |    -2.609 us |  -0.67% |
|     64      |    4096    |  to_lower  | 415.424 us | 418.310 us |     2.886 us |   0.69% |
|     128     |    4096    |  to_lower  | 497.007 us | 483.901 us |   -13.106 us |  -2.64% |
|     256     |    4096    |  to_lower  | 641.068 us | 637.852 us |    -3.217 us |  -0.50% |
|     512     |    4096    |  to_lower  |   1.091 ms |   1.122 ms |    30.222 us |   2.77% |
|    1024     |    4096    |  to_lower  |   1.794 ms |   1.835 ms |    41.097 us |   2.29% |
|     32      |   32768    |  to_lower  | 466.142 us | 437.515 us |   -28.627 us |  -6.14% |
|     64      |   32768    |  to_lower  | 555.567 us | 545.410 us |   -10.157 us |  -1.83% |
|     128     |   32768    |  to_lower  | 767.102 us | 754.029 us |   -13.073 us |  -1.70% |
|     256     |   32768    |  to_lower  |   1.193 ms |   1.145 ms |   -47.647 us |  -3.99% |
|     512     |   32768    |  to_lower  |   1.991 ms |   1.916 ms |   -75.124 us |  -3.77% |
|    1024     |   32768    |  to_lower  |   3.598 ms |   3.510 ms |   -88.139 us |  -2.45% |
|     32      |   262144   |  to_lower  |   1.309 ms |   1.305 ms |    -3.630 us |  -0.28% |
|     64      |   262144   |  to_lower  |   3.011 ms |   2.967 ms |   -44.315 us |  -1.47% |
|     128     |   262144   |  to_lower  |   8.594 ms |   8.618 ms |    23.580 us |   0.27% |
|     256     |   262144   |  to_lower  |  24.020 ms |  23.890 ms |  -130.103 us |  -0.54% |
|     512     |   262144   |  to_lower  |  54.555 ms |  54.901 ms |   346.397 us |   0.63% |
|    1024     |   262144   |  to_lower  | 116.653 ms | 117.204 ms |   551.282 us |   0.47% |
|     32      |  2097152   |  to_lower  |   8.188 ms |   8.236 ms |    47.144 us |   0.58% |
|     64      |  2097152   |  to_lower  |  22.978 ms |  22.333 ms |  -644.958 us |  -2.81% |
|     128     |  2097152   |  to_lower  |  71.841 ms |  70.392 ms | -1449.624 us |  -2.02% |
|     256     |  2097152   |  to_lower  | 208.806 ms | 207.623 ms | -1182.731 us |  -0.57% |
|     512     |  2097152   |  to_lower  | 480.379 ms | 479.444 ms |  -935.100 us |  -0.19% |
|     32      |  16777216  |  to_lower  |  61.061 ms |  62.283 ms |     1.221 ms |   2.00% |
|     64      |  16777216  |  to_lower  | 177.504 ms | 172.959 ms | -4544.705 us |  -2.56% |

```

**Pull request merged**